File: Java/Matrix Validation/Validation of Involutory matrix.java
class involutory_matrix_validation {
    int[][] matrixMultiply(int[][] A, int[][] B){
        int[][] sum = new int[A.length][B[0].length];
        int calc;
        for (int row = 0; row < A.length; row++) {
            for (int col = 0; col < B[row].length; col++) {
                calc = 0;
                for (int m = 0; m < B.length; m++) {
                    calc += A[row][m] * B[m][col];
                }
                sum[row][col] = calc;
            }
        }
        return sum;
    }
    boolean validIdentityMatrix(int[][] matrix){
        boolean flag = true; int sameCount = 0;
        if(matrix.length == matrix[0].length){
            for (int row = 0; row < matrix.length; row++) {
                for (int col = 0; col < matrix[row].length; col++) {
                    if((row == col && row != 0 && col != 0) && (matrix[(row-1)][(col-1)] == matrix[row][col])){
                        sameCount++;
                    }else if(row != col && matrix[row][col] != 0){
                        flag = false;
                    }
                }
            }
            if(flag == true && (sameCount+1) == matrix.length){
                return true;
            }else{
                return false;
            }
        }else{
            return false;
        }
    }
    boolean validInvolutoryMatrix(int[][] matrix, int[][] result){
        if(validIdentityMatrix(result)){
            matrix = matrixMultiply(matrix, matrix);
            if(matrix.length == result.length && matrix[0].length == result[0].length){
                for (int row = 0; row < matrix.length; row++) {
                    for (int col = 0; col < matrix[row].length; col++) {
                        if(matrix[row][col] != result[row][col])
                            return false;
                    }
                }
            }
        }else{
            return false;
        }
        return true;
    }
    public static void main(String[] args) {
        // Involutory matrix if matrix multiply by itself return the identity matrix. (A^2 = I)
        int[][] matrix = {
            { 2, 3 },
            {-1,-2 }
        };
        int[][] result = {
            { 1, 0 },
            { 0, 1 }
        };

        involutory_matrix_validation get = new involutory_matrix_validation();
        if(get.validInvolutoryMatrix(matrix, result) == true){
            System.out.println("Valid Involutory matrix");
        }else{
            System.out.println("Invalid Involutory matrix");
        }
    }
}

File: Python/Matrix Validation/Validation of Involutory matrix.py
def matrixMultiply(A, B):
    result = [[None]*len(B[0]) for row in range(len(A))]
    for row in range(0, len(A)):
        for col in range(0, len(B[row])):
            calc = 0
            for m in range(0, len(B)):
                calc += A[row][m] * B[m][col]
            result[row][col] = calc
    return result


def validIdentityMatrix(matrix):
    flag = True; sameCount = 0
    if len(matrix) == len(matrix[0]):
        for row in range(0, len(matrix)):
            for col in range(0, len(matrix[row])):
                if (row == col and row != 0 and col != 0) and (matrix[(row-1)][(col-1)] == matrix[row][col]):
                    sameCount += 1
                elif row != col and matrix[row][col] != 0:
                    flag = False

        if flag == True and (sameCount+1) == len(matrix):
            return True
        else:
            return False
    else:
        return False

        
def validInvolutoryMatrix(matrix, result):
    if validIdentityMatrix(result):
        matrix = matrixMultiply(matrix, matrix);
        if len(matrix) == len(result) and len(matrix[0]) == len(result[0]):
            for row in range(0, len(matrix)):
                for col in range(0, len(matrix[row])):
                    if matrix[row][col] != result[row][col]:
                        return False
    else:
        return False
    return True

# Involutory matrix if matrix multiply by itself return the identity matrix. (A^2 = I)
matrix = [[ 2,  3],
          [-1, -2]]
result = [[1, 0],
          [0, 1]]

if validInvolutoryMatrix(matrix, result) == True:
    print("Valid Involutory matrix")
else:
    print("Invalid Involutory matrix")